great_expectations/execution_engine/pandas_execution_engine.py

```diff
@@ -163,15 +181,39 @@ def get_batch_data_and_markers(
             buf = BytesIO(s3_object["Body"].read())
             buf.seek(0)
             df = reader_fn(buf, **reader_options)
+
+        elif isinstance(batch_spec, AzureBatchSpec):
+            if self._azure is None:
+                raise ge_exceptions.ExecutionEngineError(
+                    f"""PandasExecutionEngine has been passed a AzureBatchSpec,
+                        but the ExecutionEngine does not have an Azure client configured. Please check your config."""
+                )
+            azure_engine: BlobServiceClient = self._azure
+            azure_url = AzureUrl(batch_spec.path)
+            reader_method: str = batch_spec.reader_method
+            reader_options: dict = batch_spec.reader_options or {}
+            blob_client: BlobClient = azure_engine.get_blob_client(
+                container=azure_url.container, blob=azure_url.blob
+            )
+            azure_object: StorageStreamDownloader = blob_client.download_blob()
+            logger.debug(
+                f"Fetching Azure blob. Container: {azure_url.container} Blob: {azure_url.blob}"
+            )
+            reader_fn = self._get_reader_fn(reader_method, azure_url.blob)
+            buf = BytesIO(azure_object.readall())
+            buf.seek(0)
```

Comment:
This all works swimmingly when I use it on our integration test for `ConfiguredAssetAzureDataConnector` but how do we test this here? Mocks can only do so much and I'm in a bit torn on how much to test in this PR.

Reply:
Would the end-to-end test that is part of docs help with this?

Reply:
UPDATE: Wrote basic tests per our previous conversation and will test actual functionality of the low-level `BytesIO` in our integration test. The logic here is identical to what we've done for S3.